src/vs/editor/common/services/languagesRegistry.ts

```diff
@@ -128,7 +128,9 @@ export class LanguagesRegistry {
 		if (Array.isArray(lang.extensions)) {
 			for (let extension of lang.extensions) {
 				mime.registerTextMime({ id: langId, mime: primaryMime, extension: extension }, this._warnOnOverwrite);
-				resolvedLanguage.extensions.push(extension);
+				if (!resolvedLanguage.extensions.includes(extension)){
```

Comment:
we generally try to make minimal changes to the VS Code under the ../vs/ directory to make it easier to take VS Code source code refreshes.  I wonder if there is a way to fix this issue without requiring changes to the forked VS Code source code, such as not registering SQL twice (assuming that's what's happening here).  Let me take a look and see what's happening here.

Reply:
This looks like a change that should go straight to vscode sources, unless theres a reason they aren't.